### PR TITLE
Helicopters not ringing, and other audio stuff

### PR DESCRIPTION
**Pull request by theR4K**

### CODE REVIEW

File: src/audio/AudioLogic.cpp [hunk 1]
```diff
@@ -2259,34 +2265,36 @@ cAudioManager::HasAirBrakes(int32 model) const
 		|| model == COACH || model == PACKER || model == FLATBED;
 }
 
-bool
+void
```
Comment (Sergeanur) [inline src/audio/AudioLogic.cpp:2268]: This returns bool

File: src/audio/AudioLogic.cpp [hunk 2]
```diff
@@ -3016,62 +3043,414 @@ cAudioManager::ProcessBoatMovingOverWater(cVehicleParams *params)
...
-		else
-			emittingVol = gHeliSfxRanges[i].m_bBaseVolume;
 
-		m_sQueueSample.m_nVolume = ComputeVolume(emittingVol, gHeliSfxRanges[i].m_fMaxDistance, m_sQueueSample.m_fDistance);
+
+	//sound on long distances
+	if (m_sQueueSample.m_fDistance >= 40.0f)
+		emittingVol = propellerSpeed * 75.0f;
+	else if (m_sQueueSample.m_fDistance >= 25.0f)
+		emittingVol = (m_sQueueSample.m_fDistance - 25.0f) * 0.06666667f * (75.0f * propellerSpeed);
```
Comment (Sergeanur) [inline src/audio/AudioLogic.cpp:3113]: `* 0.06666667f` is `/ 15.0f`

File: src/audio/AudioLogic.cpp [hunk 3]
```diff
@@ -3016,62 +3043,414 @@ cAudioManager::ProcessBoatMovingOverWater(cVehicleParams *params)
...
+			m_sQueueSample.m_fSoundIntensity = SOUND_INTENSITY;
+			m_sQueueSample.m_bReleasingSoundFlag = false;
+			m_sQueueSample.m_nReleasingVolumeDivider = 5;
+			m_sQueueSample.m_bReverbFlag = true;
+			m_sQueueSample.m_bRequireReflection = false;
+			AddSampleToRequestedQueue();
+		}
+	}
+
+	if (params->m_fDistance >= 19600.0f)
```
Comment (Sergeanur) [inline src/audio/AudioLogic.cpp:3148]: SQ(140.0f)

File: src/audio/AudioLogic.cpp [hunk 4]
```diff
@@ -3016,62 +3043,414 @@ cAudioManager::ProcessBoatMovingOverWater(cVehicleParams *params)
...
+			m_sQueueSample.m_bRequireReflection = false;
+			AddSampleToRequestedQueue();
+		} else if (boat != nil) {
+			m_sQueueSample.m_nSampleIndex = SFX_SEAPLANE_PRO1;
+			m_sQueueSample.m_nBankIndex = SFX_BANK_0;
+
+			if (accelerateState <= 0 && brakeState <= 0)
+				m_sQueueSample.m_nFrequency = 14287 + Min(1.0f, freqModifier) * 3713;
+			else
+				m_sQueueSample.m_nFrequency = 18000 + Min(1.0f, freqModifier * (Max(accelerateState, brakeState) / 255.0f)) * 2204;
```
Comment (Sergeanur) [inline src/audio/AudioLogic.cpp:3193]: Looks like this if-else should be reversed

File: src/audio/AudioLogic.cpp [hunk 5]
```diff
@@ -3016,62 +3043,414 @@ cAudioManager::ProcessBoatMovingOverWater(cVehicleParams *params)
...
+			if (propellerSpeed < 1.0f)
+				m_sQueueSample.m_nFrequency = (propellerSpeed + 1.0f) * (m_sQueueSample.m_nFrequency / 2.0f);
+			m_sQueueSample.m_nFrequency = clamp(m_sQueueSample.m_nFrequency, freqFrontPrev - 197, freqFrontPrev + 197);
```
Comment (Sergeanur) [inline src/audio/AudioLogic.cpp:3196]: clamp isn't what originally done here, right? Best to keep original way then

File: src/audio/AudioLogic.cpp [hunk 6]
```diff
@@ -3016,62 +3043,414 @@ cAudioManager::ProcessBoatMovingOverWater(cVehicleParams *params)
...
+			m_sQueueSample.m_nBankIndex = SFX_BANK_0;
+			m_sQueueSample.m_nFrequency = (volumeModifier + 1) * 16000 + freq;
+		} else if (boat) {
+			m_sQueueSample.m_nSampleIndex = SFX_SEAPLANE_PRO2;
+			m_sQueueSample.m_nBankIndex = SFX_BANK_0;
+
+			if (accelerateState <= 0 && brakeState <= 0)
+				m_sQueueSample.m_nFrequency = 7143 + Min(1.0f, freqModifier) * 1857;
+			else
+				m_sQueueSample.m_nFrequency = 9000 + Min(1.0f, (Max(accelerateState, brakeState) / 255) * freqModifier) * 1102;
```
Comment (Sergeanur) [inline src/audio/AudioLogic.cpp:3249]: Again reversed if-else

File: src/audio/AudioLogic.cpp [hunk 7]
```diff
@@ -3016,62 +3043,414 @@ cAudioManager::ProcessBoatMovingOverWater(cVehicleParams *params)
...
+		}
+	}
+
+
+	if (boat) {
+		if (TheCamera.Cams[TheCamera.ActiveCam].Mode == CCam::MODE_FIXED && m_sQueueSample.m_fDistance < 20.0f && propellerSpeed > 0.0f) {
+			m_sQueueSample.m_nVolume = ComputeVolume(propellerSpeed * 100.0f, 20.0f, m_sQueueSample.m_fDistance);
+			if (m_sQueueSample.m_nVolume) {
+
+				if (accelerateState <= 0 && brakeState <= 0)
```
Comment (Sergeanur) [inline src/audio/AudioLogic.cpp:3319]: and again

File: src/audio/AudioLogic.cpp [hunk 8]
```diff
@@ -3016,62 +3043,414 @@ cAudioManager::ProcessBoatMovingOverWater(cVehicleParams *params)
...
+		distanceOld = params->m_fDistance;
+		
+		if (automobile != nil)
+			automobile->GetComponentWorldPosition(CAR_BOOT, m_sQueueSample.m_vecPos);
+		else if (params->m_VehicleType == VEHICLE_TYPE_HELI) 
+			m_sQueueSample.m_vecPos = { 0.0f, -10.0f, 0.0f }; //this is from android, but for real it's not used
+
+		params->m_bDistanceCalculated = false;
+		params->m_fDistance = GetDistanceSquared(m_sQueueSample.m_vecPos);
+		if (params->m_fDistance < 729.0f) {
```
Comment (Sergeanur) [inline src/audio/AudioLogic.cpp:3359]: SQ(27)

File: src/audio/AudioLogic.cpp [hunk 9]
```diff
@@ -4558,20 +4936,107 @@ cAudioManager::ProcessWaterCannon(int32)
 	}
 }
 
+//positon of arcade machines
+CVector aVecExtraSoundPosition[] = { {-1042.546f, 88.793999f, 11.324f}, {-1004.476f, 181.69701f, 11.324f} };
```
Comment (Sergeanur) [inline src/audio/AudioLogic.cpp:4940]: 88.793999f is 88.794f and 181.69701f is 181.697f

**Review by Sergeanur — CHANGES_REQUESTED** [review]

File: src/render/Fluff.h [hunk 10]
```diff
@@ -103,6 +103,8 @@ class CEscalator
 	void Update(void);
 	void SwitchOff(void);
 	void AddThisOne(CVector pos0, CVector pos1, CVector pos2, CVector pos3, bool b_isMovingDown);
+	bool IsActive() { return m_bIsActive; };
+	CVector GetPosition() { return m_midPoint; };
```
Comment (Sergeanur) [inline src/render/Fluff.h:107]: const CVector &

File: src/render/Fluff.h [hunk 11]
```diff
@@ -114,6 +116,7 @@ class CEscalators
 	static void Update(void);
 	static void AddOne(CVector pos0, CVector pos1, CVector pos2, CVector pos3, bool b_isMovingDown);
 	static void Shutdown(void);
+	static CEscalator GetEscalator(int ind) { return aEscalators[ind]; };
```
Comment (Sergeanur) [inline src/render/Fluff.h:119]: const CEscalator &

File: src/audio/AudioLogic.cpp (hunk 5 above)
Comment (theR4K) [reply src/audio/AudioLogic.cpp:3196]: originally hire used
if(freq < freqFrontPrev){
   if(freq < freqFrontPrev - 197)
       freq = freqFrontPrev - 197;
}else{
    if(freq > freqFrontPrev + 197)
        freq = freqFrontPrev + 197;
}
but this is clear clamp. But it's use one more compare....

Comment (Sergeanur) [reply src/audio/AudioLogic.cpp:3196]: Yeah the first check shows that it wasn't clamp macro

Comment (theR4K) [reply src/audio/AudioLogic.cpp:3196]: but this is it

Comment (theR4K) [reply src/audio/AudioLogic.cpp:3196]: all this code looks like macro, bcs Its repeating 3 times

File: src/audio/AudioLogic.cpp (hunk 1 above)
Comment (theR4K) [reply src/audio/AudioLogic.cpp:2268]: this bool means nothing, bcs it false only if distance too big

File: src/render/Fluff.h (hunk 11 above)
Comment (theR4K) [reply src/render/Fluff.h:119]: cant be const

Comment (ShFil119) [reply src/render/Fluff.h]: reference

File: src/audio/AudioLogic.cpp (hunk 5 above)
Comment (Sergeanur) [reply src/audio/AudioLogic.cpp:3213]: But it still ain't clamp macro

File: src/render/Fluff.h (hunk 11 above)
Comment (Sergeanur) [reply src/render/Fluff.h]: Is it being changed in audio code or why it can't be const?

File: src/audio/AudioLogic.cpp [hunk 12]
```diff
@@ -3016,62 +3060,414 @@ cAudioManager::ProcessBoatMovingOverWater(cVehicleParams *params)
...
+			if (bike->m_wheelStatus[i] == WHEEL_STATUS_BURST && bike->m_aWheelTimer[i] > 0.0f)
+				wheelBurst = true;
+		if (!wheelBurst)
+			return;
+		break;
+	default:
+		return;
+	}
+	modifier = Min(1.0f, Abs(params->m_fVelocityChange) / (0.3f * params->m_pTransmission->fMaxVelocity));
+	if (modifier > 0.01) { //mb can be replaced by (emittingVol > 1)
```
Comment (ShFil119) [inline src/audio/AudioLogic.cpp:3444]: `0.01f`

File: src/audio/AudioLogic.cpp [hunk 13]
```diff
@@ -3016,62 +3060,414 @@ cAudioManager::ProcessBoatMovingOverWater(cVehicleParams *params)
...
+		emittingVol = (100.0 * modifier);
```
Comment (ShFil119) [inline src/audio/AudioLogic.cpp:3445]: also

File: src/audio/AudioLogic.cpp [hunk 14]
```diff
@@ -4558,20 +4953,107 @@ cAudioManager::ProcessWaterCannon(int32)
...
+	const float SOUND_INTENSITY = 18.0f;
+	const uint8 EMITTING_VOLUME = 50;
+
+	float distance;
+
+	for (int i = 0; i < ARRAY_SIZE(aVecExtraSoundPosition); i++) {
+		m_sQueueSample.m_vecPos = aVecExtraSoundPosition[i];
+		distance = GetDistanceSquared(m_sQueueSample.m_vecPos);
+		if (distance < SQR(SOUND_INTENSITY)) {
+			if (distance > 0.0)
```
Comment (ShFil119) [inline src/audio/AudioLogic.cpp:4971]: +1

File: src/render/Fluff.h (hunk 11 above)
Comment (theR4K) [reply src/render/Fluff.h]: ask compiler... I'm C# developer)

File: src/audio/AudioLogic.cpp [hunk 15]
```diff
@@ -2631,6 +2657,24 @@ cAudioManager::ProcessVehicleOneShots(cVehicleParams *params)
 			emittingVol = m_anRandomTable[4] % 25 + 75;
 			break;
 		}
+		case SOUND_31:{
+			const float SOUND_INTENSITY = 35.0f;
```
Comment (erorcun) [inline src/audio/AudioLogic.cpp:2661]: heliIndex should be declared in here

File: src/audio/AudioLogic.cpp [hunk 16]
```diff
@@ -3016,62 +3060,414 @@ cAudioManager::ProcessBoatMovingOverWater(cVehicleParams *params)
...
+	} else {
+		//vacuum cleaner sound
+		vecPosOld = m_sQueueSample.m_vecPos;
+		distanceCalculatedOld = params->m_bDistanceCalculated;
+		distanceOld = params->m_fDistance;
+		
+		if (automobile != nil)
+			automobile->GetComponentWorldPosition(CAR_BOOT, m_sQueueSample.m_vecPos);
+		else if (params->m_VehicleType == VEHICLE_TYPE_HELI) 
+			m_sQueueSample.m_vecPos = { 0.0f, -10.0f, 0.0f }; //this is from android, but for real it's not used
```
Comment (erorcun) [inline src/audio/AudioLogic.cpp:3372]: we don't prefer new C++ things

File: src/audio/AudioLogic.cpp [hunk 17]
```diff
@@ -3016,62 +3060,414 @@ cAudioManager::ProcessBoatMovingOverWater(cVehicleParams *params)
...
+	if (emittingVol != 0) {
+		m_sQueueSample.m_nVolume = ComputeVolume(emittingVol, SOUND_INTENSITY, m_sQueueSample.m_fDistance);
 		if (m_sQueueSample.m_nVolume != 0) {
-			m_sQueueSample.m_nCounter = i + 65;
-			m_sQueueSample.m_nSampleIndex = i + SFX_HELI_1;
+			m_sQueueSample.m_nCounter = 88;
+			if (boat != nil) {
+				m_sQueueSample.m_nSampleIndex = SFX_SEAPLANE_PRO3;
+				m_sQueueSample.m_nBankIndex = SFX_BANK_0;
+				if (accelerateState <= 0 && brakeState <= 0)
```
Comment (Sergeanur) [inline src/audio/AudioLogic.cpp:3140]: unreversed check

**Pull request merged**